### DISCUSSION TITLE
Increase speed without refresh

### DISCUSSION THREAD

**Discussion by umbert0dev** [matteobruni/tsparticles#5426] [Q&A] [answered]
Hi there i m looking for a way to increase speed without refresh the entire component.
I'm using this to change the speed but it is not what i'm looking for because i'd like see the particles speed up
```
particlesContainer.options.particles.move.speed = newSpeed;
particlesContainer.refresh();
```
Thank you

**Comment by matteobruni** [comment]
> Hi there i m looking for a way to increase speed without refresh the entire component. I'm using this to change the speed but it is not what i'm looking for because i'd like see the particles speed up
> 
> ```
> particlesContainer.options.particles.move.speed = newSpeed;
> particlesContainer.refresh();
> ```
> 
> Thank you

You need to create a custom plugin, options are cached and a change to that object requires a refresh. If you want to handle speed differently you need to create a custom plugin, you can start from the existing `move-base` plugin that you can find here: https://github.com/tsparticles/tsparticles/tree/main/move/base

**Comment by umbert0dev** [comment, marked as answer]
thank you so much for the answer. I made a simple custom plugin and it worked. I share my solution (i'm using vuejs)
```
import { type IParticleMover } from "@tsparticles/engine";
export default class DynamicSpeedMover implements IParticleMover {
    speedRef = 0;
    constructor(speedRef) {
        this.speedRef = speedRef;
    }

    init(particle) {
    }

    isEnabled(particle) {
        return !particle.destroyed && particle.options.move.enable;
    }

    move(particle, delta) {
        if (!particle.options.move.enable) {
            return;
        }
        const container = particle.container;
        const pxRatio = container.retina.pixelRatio;
        const speed = this.speedRef.value * pxRatio;
        particle.retina.moveSpeed = speed;
    }
}
```
```
const speedRef = ref(3); // initial speed
app.provide("speedRef", speedRef); // pass speed reference to main component to change it dynamically (i made it with a slider)
app.use(Particles, {
  init: async engine => {
    await loadFull(engine);
    await engine.addMover(
      "DynamicSpeedMover",
      () => {
        return Promise.resolve(new DynamicSpeedMover(speedRef));
      },
      true
    )
  },
});
```
thank you
